augment run script specification with executable passed as parameter (#241)
MicroEJ/docs: PlatformDeveloperGuide/platformCreation.rst

This script is required only for Platforms intended to run a :ref:`MicroEJ Testsuite <platform_testsuite>` on device.

> The BSP run script is responsible to invoke a third-party tool to upload and start the executable file (``application.out``) on device.

Reviewer: ```suggestion
The BSP run script is responsible to invoke a third-party tool to upload and start the executable file on device.
```

Remove mention of `application.out` as the executable can be named differently is specified by a CLI parameter.

The run script must implement the following specification:

- On Windows operating system, it is a Windows batch file named ``run.bat``.
- On Mac OS X or Linux operating systems, it is a shell script named ``run.sh``, with execution permission enabled.

> - The executable file is passed as first script parameter if there is one, otherwise it is the ``application.out`` file located in the directory from where the script has been executed.

Reviewer: ```suggestion
- The executable file is passed as first script parameter; if no parameter is passed, the executable file is the ``application.out`` file located in the directory from where the script has been executed (the current directory).
```

Not sure on the formulation, here is a rephrasing.
